services/src/main/java/org/keycloak/protocol/oidc/endpoints/TokenEndpoint.java

```diff
@@ -1318,6 +1325,7 @@ public Response permissionGrant() {
         List<String> permissions = formParams.get("permission");
 
         if (permissions != null) {
+            event.detail(Details.PERMISSION, String.join("|", permissions));
```

Comment:
I promise this is the last comment from me :)

Do you really need to audit this every time?

Reply:
The intention here is to have enough data in the audit records should a deep "who is doing what and how" audit takes place, to prove accountability of access to resources.

Consider a company with (very) sensitive resources, where it is critical that only the right users with the right permissions can consume them.

Reply:
Yeah, I see that. I'm more concerned about performance and doing this every time.

I'm going to accept as-is for now and keep here a note that we might review this later if it becomes an issue. If people are not willing to sacrifice performance for a better audit/security we should probably think about changing the authorization settings for a client to configure how the audit should behave.